File: src/main/java/org/mitallast/queue/rest/RestRequest.java
package org.mitallast.queue.rest;

import io.netty.buffer.ByteBuf;
import io.netty.handler.codec.http.HttpMethod;

import java.util.Map;

public interface RestRequest {
    HttpMethod getHttpMethod();

    String getQueryPath();

    String getUri();

    Map<String, String> getParamMap();

    String param(String param);

    boolean hasParam(String param);

    ByteBuf content();

    ResponseBuilder response();
}

File: src/main/java/org/mitallast/queue/rest/RestRequest.kt
package org.mitallast.queue.rest

import io.netty.buffer.ByteBuf
import io.netty.handler.codec.http.HttpMethod
import javaslang.collection.Map

interface RestRequest {
    val httpMethod: HttpMethod

    val queryPath: String

    val uri: String

    val paramMap: Map<String, String>

    val content: ByteBuf

    fun param(param: String): String

    fun hasParam(param: String): Boolean

    fun response(): ResponseBuilder
}
